charts/karmada-dashboard/templates/karmada-dashboard-web-service.yaml

```diff
@@ -12,5 +12,8 @@ spec:
       targetPort: http
       protocol: TCP
       name: http
+      {{- with .Values.web.service.nodePort }}
```

Comment:
here we need more robust check, if user input 
```yaml
service:
    type: ClusterIP
    port: 8000
    nodePort: 32000
```
the render output will be :
```yaml
spec:
  type: ClusterIP
  ports:
    - port: 8000
      targetPort: http
      protocol: TCP
      name: http
      nodePort: 32000
```

so what about 
```
{{- if and (eq .Values.web.service.type "NodePort") (not (empty .Values.web.service.nodePort)) }}
nodePort: {{ .Values.web.service.nodePort }}
{{- end }}
```

Reply:
Updated